### ISSUE TITLE
Select icon prop cannot accept an object

### ISSUE THREAD

**Issue by robcaldecottvelo** [jamesmfriedman/rmwc#562]
RMWC 5.7.2

The `Select` `icon` prop cannot accept an `object` which causes issues if you want to use a custom `strategy`. Attempting to use an object will cause the following error:

```
Objects are not valid as a React child (found: object with keys {icon}).  If you meant to render a collection of children, use an array instead.
    in div (created by SelectRoot)
    in SelectRoot (created by ForwardRef)
    in Ripple (created by ForwardRef)
    in ForwardRef (created by Context.Consumer)
    in withRipple(SelectRoot) (created by SelectBase)
    in SelectBase (created by Select)
```

Example here:

https://codesandbox.io/s/select-icon-bug-bpcy2

Note that you can workaround this issue by wrapping the `Select` in an `RMWCProvider` that specifies the correct icon props to use.

**Comment by jamesmfriedman** [comment]
I can confirm that this is an issue in 5.x and also that it is fixed in the upcoming release.

**Comment by jamesmfriedman** [comment]
Fixed in 6.0.0
